packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingView.tsx

```diff
@@ -139,10 +139,12 @@ export function TrendingView(props: TrendingViewProps) {
     useEffect(() => setTabIndex(0), [networkType])
     // #endregion
     // #region merge trending
+
     const { value: { trending } = {}, loading: loadingTrending } = useTrendingById(
+        result.pluginID,
         result.type === SearchResultType.FungibleToken
             ? result.id || searchedContractAddress || ''
-            : result.address || '',
+            : (result.pluginID === NetworkPluginID.PLUGIN_SOLANA ? result.name : result.address) || '',
```

Comment:
Maybe I can add the  `id` field on NonFungibleCollection from data source.

Reply:
handled on api layer instead.